MSTP Documentation: Enhanced, corrected and refined the existing content

vinaykumar-kayyur opened the pull request:
Improved the MSTP documentation by adding new content, made few technical corrections as per MSTP standard and refined existing descriptions for better readability and grammatical accuracy.

mssonicbld commented:
/azp run

jarias-lfx commented:
/easycla

mssonicbld commented:
/azp run

vinaykumar-kayyur commented:
@ridahanif96 @divyachandralekha @adyeung Kindly review and approve the MSTP documentation changes

vinaykumar-kayyur commented:
Hello @adyeung, a gentle reminder! Can you please help get this reviewed at the earliest?

vinaykumar-kayyur commented:
Hello @adyeung, another gentle reminder! Kindly help get this reviewed at the earliest...

divyachandralekha commented on doc/MSTP/MSTP.md:
```diff
@@ -699,15 +682,15 @@ Following debug commands will be supported for enabling additional logging which
+The following debug commands will be supported for displaying internal data structures:
 
-- debug spanning_tree dump global 
+- debug spanning_tree dump mst global
```
Please remove 'mst' keyword

vinaykumar-kayyur replied:
If we remove the 'mst' keyword here, it would be not in line with the MST instance specific debug command defined below. Please clarify if that's correct as per the intended design.

divyachandralekha replied:
We plan to use the existing commands, and if MSTP requires more specific information, those additional commands are introduced with the 'mst' keyword

vinaykumar-kayyur replied:
Have addressed this comment. Kindly check.

divyachandralekha commented on doc/MSTP/MSTP.md:
````diff
@@ -392,135 +395,129 @@ MSTP standard does not support uplink fast so uplink fast functionality will be
+    The output of this command would be as follows for 'mstp':
 ```
-admin@sonic: show spanning_tree detail
+admin@sonic: show spanning_tree mst
````
Please remove 'mst' keyword

vinaykumar-kayyur replied:
This change was done to be inline with this command header above. Please clarify if the 'mst' keyword needs to be removed at both the places and is as per the intended design.

divyachandralekha replied:
We plan to use the existing commands, and if MSTP requires more specific information, those additional commands are introduced with the 'mst' keyword

vinaykumar-kayyur replied:
Have addressed this comment. Kindly check.

divyachandralekha commented on doc/MSTP/MSTP.md:
```diff
@@ -668,26 +654,23 @@ Ethernet46  128    800     N        FORWARDING    Root     0             3276800
-
 ## Clear Commands
-- sonic-clear spanning_tree statistics 
+- sonic-clear spanning_tree mst statistics
```
Please remove 'mst' keyword

vinaykumar-kayyur replied:
If we remove the 'mst' keyword here, it would be not in line with the MST instance specific clear command defined below. Please clarify if that's correct as per the intended design.

divyachandralekha replied:
We plan to use the existing commands, and if MSTP requires more specific information, those additional commands are introduced with the 'mst' keyword

vinaykumar-kayyur replied:
Have addressed this comment. Kindly check.

vinaykumar-kayyur commented on doc/MSTP/MSTP.md:
````diff
@@ -530,15 +527,14 @@ Root                 Address 8000.80a2.3526.0c5e
+PortChannel1001    DESIGNATED   FORWARDING      1000       128.45       P2P
 ```
 
-- show spanning_tree  mst detail 
````
The command header declared in this line have 'mst' keyword, but the command on line 540 below doesnt include it. Please clarify what is the correct command here?

divyachandralekha replied:
Please don't change the command format that we have already updated. The code changes align with what is outlined in the HLD. Our intention is to use the existing commands for MSTP as well. Any commands added specifically for MSTP will include the 'mst' keyword

vinaykumar-kayyur replied:
Have addressed this comment. Kindly check.

divyachandralekha commented on doc/MSTP/MSTP.md:
```diff
@@ -392,135 +395,129 @@ MSTP standard does not support uplink fast so uplink fast functionality will be
  
 ## Show Commands
 
 - show spanning_tree mst
```
Please don't change the command format that we have already updated. The code changes align with what is outlined in the HLD. Our intention is to use the existing commands for MSTP as well. Any commands added specifically for MSTP will include the 'mst' keyword

vinaykumar-kayyur replied:
Have addressed this comment. Kindly check.

mssonicbld commented:
/azp run

vinaykumar-kayyur commented:
@divyachandralekha, Thank you for your kind clarification! 

I have addressed your comments now and updated this PR. Kindly review and provide your approval if no further comments to address.

vinaykumar-kayyur commented:
@divyachandralekha, another gentle reminder to complete the review and provide your approval for the merge at the earliest.

vinaykumar-kayyur commented:
Thank you @divyachandralekha for the approval! Request @adyeung to get this PR upstreamed into the mainline at the earliest.

vinaykumar-kayyur commented:
Hello @adyeung ... A gentle reminder to get this PR committed to the mainline... Kindly do the needful at the earliest...

vinaykumar-kayyur commented:
Hello @adyeung ... Another gentle reminder to get this PR committed to the mainline at the earliest!

vinaykumar-kayyur commented:
Hello @adyeung ... Another gentle reminder! Kindly get this PR committed to the mainline at the earliest!

vinaykumar-kayyur commented:
Hello @adyeung... Another gentle reminder to kindly do the needful at the earliest!
